analytics.go

```diff
@@ -157,36 +164,94 @@ func (c *client) Enqueue(msg Message) (err error) {
 		m.Type = "alias"
 		m.MessageId = makeMessageId(m.MessageId, id)
 		m.Timestamp = makeTimestamp(m.Timestamp, ts)
+		if m.Context == nil {
+			m.Context = &Context{}
+			m.Context.Library = LibraryInfo{
+				Name:    "analytics-go",
+				Version: "1.0.0",
+			}
+		}
+
 		msg = m
 
 	case Group:
 		m.Type = "group"
 		m.MessageId = makeMessageId(m.MessageId, id)
+		if m.AnonymousId == "" {
+			m.AnonymousId = m.UserId
+		}
 		m.Timestamp = makeTimestamp(m.Timestamp, ts)
+		if m.Context == nil {
+			m.Context = &Context{}
+			m.Context.Library = LibraryInfo{
+				Name:    "analytics-go",
+				Version: "1.0.0",
+			}
+		}
 		msg = m
 
 	case Identify:
 		m.Type = "identify"
 		m.MessageId = makeMessageId(m.MessageId, id)
+		if m.AnonymousId == "" {
+			m.AnonymousId = m.UserId
+		}
 		m.Timestamp = makeTimestamp(m.Timestamp, ts)
+		if m.Context == nil {
+			m.Context = &Context{}
+			m.Context.Library = LibraryInfo{
+				Name:    "analytics-go",
+				Version: "1.0.0",
+			}
+		}
 		msg = m
 
 	case Page:
 		m.Type = "page"
 		m.MessageId = makeMessageId(m.MessageId, id)
+		if m.AnonymousId == "" {
+			m.AnonymousId = m.UserId
+		}
 		m.Timestamp = makeTimestamp(m.Timestamp, ts)
+		if m.Context == nil {
+			m.Context = &Context{}
+			m.Context.Library = LibraryInfo{
+				Name:    "analytics-go",
+				Version: "1.0.0",
+			}
+		}
 		msg = m
 
 	case Screen:
 		m.Type = "screen"
 		m.MessageId = makeMessageId(m.MessageId, id)
+		if m.AnonymousId == "" {
+			m.AnonymousId = m.UserId
+		}
 		m.Timestamp = makeTimestamp(m.Timestamp, ts)
+		if m.Context == nil {
+			m.Context = &Context{}
+			m.Context.Library = LibraryInfo{
+				Name:    "analytics-go",
+				Version: "1.0.0",
+			}
+		}
 		msg = m
 
 	case Track:
 		m.Type = "track"
 		m.MessageId = makeMessageId(m.MessageId, id)
+		if m.AnonymousId == "" {
+			m.AnonymousId = m.UserId
+		}
 		m.Timestamp = makeTimestamp(m.Timestamp, ts)
+		if m.Context == nil {
```

Comment:
Instead of repeating this again, let's write another function like others makeContext() and makeAnonymousId()

Reply:
Changes done and committed